src/Network/Consul/Types.hs

```diff
@@ -40,12 +40,13 @@ import Data.Word
 import Network.HTTP.Client (Manager)
 import Network.Socket
 
-data ConsulClient = ConsulClient{
-  ccManager :: Manager,
-  ccHostname :: Text,
-  ccPort :: PortNumber,
-  ccWithTls :: Bool
-}
+data ConsulClient = ConsulClient
+  { ccManager :: Manager
+  , ccHostname :: Text
+  , ccPort :: PortNumber
+  , ccWithTls :: Bool
+  , ccDatacenter :: Datacenter
```

Comment:
I believe it would be better to have this be `Maybe Datacenter`, because you can leave it away entirely in consul requests (and I think in most of my consul usage I do leave it away).

I also think we do not need to update the signatures of `initializeConsulClient` and `initializeTlsConsulClient` in this PR: We can default the datacenter to `Nothing` there, and users can use record updates if they need to change the datacenter (see main review comment for an example).

Reply:
Thanks for the feedback, I can make each of those updates!

Reply:
I've updated this in the latest version of the PR.

Reply:
I've applied these updates in the latest PR.